general/mem_manager.hpp

```diff
@@ -137,16 +141,18 @@ class Memory
       VALID_DEVICE  = 1 << 5, ///< %Device pointer is valid
       USE_DEVICE    = 1 << 6, /**< Internal device flag, see e.g.
                                    Vector::UseDevice() */
-      ALIAS         = 1 << 7  ///< Pointer is an alias
+      ALIAS         = 1 << 7, ///< Pointer is an alias
+      USE_TEMPORARY = 1 << 8  ///< Temporary Device memory flag
```

Comment:
We should unify `TEMP`, `TEMPORARY` to either `TEMP` or `POOL`

Reply:
I agree with that. For Umpire, `POOL` doesn't make sense because both spaces can be a pool. In our case we use one kind of Umpire Pool for the permanent device memory and another kind of pool for the temporary device memory. My vote is for `TEMP` but I'm okay with `POOL` if that's what others think is best.

The reason for the pool with permanent memory is there are still some small stack-allocated objects that own device memory and tagging all of them as temporary is a bit involved and would result in a more fragmented temporary pool, which we want to leave open for large temporary device allocations.